backend/src/main/java/com/huellapositiva/application/controller/RecoveryPasswordApiController.java

```diff
@@ -0,0 +1,85 @@
+package com.huellapositiva.application.controller;
+
+import com.huellapositiva.application.exception.UserNotFoundException;
+import com.huellapositiva.domain.actions.FetchCredentialsAction;
+import com.huellapositiva.domain.exception.TimeForRecoveringPasswordExpiredException;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+
+@Controller
+@RequestMapping("/api/v1/restore-password")
+public class RecoveryPasswordApiController {
+
+    @Autowired
+    FetchCredentialsAction credentialsAction;
+
+    @Operation(
+            summary = "Send an email to recovery password",
+            description = "Fetch the credential, create an email with a link to recover password, send the email.",
+            tags = {"contactPerson, volunteers, recoveryPassword"}
+    )
+    @ApiResponses(
+            value = {
+                    @ApiResponse(
+                            responseCode = "204",
+                            description = "No content, email sent successfully."
+                    ),
+                    @ApiResponse(
+                            responseCode = "404",
+                            description = "Not found, email not found in database."
+                    ),
+                    @ApiResponse(
+                            responseCode = "500",
+                            description = "Internal server error, could not fetch the user data due to a connectivity issue."
+                    )
+            }
+    )
+    @GetMapping("/sendRecoveryPasswordEmail/{email}")
```

Comment:
A request that causes state changes in the backed should be a POST request.
The email address of someone is very easily guessable so we must not use it in the request path, instead we should send it in the body.
This request does not require any role so it should be CSRF protected to avoid anyone making this request outside our domain

Reply:
Good morning!!! Yes, you are right. We fixed it.